### PR TITLE
fix: add storeCar to client interface.ts

### PR DESCRIPTION
**Pull request by olizilla**
I'm a ts-noob, but this feels like something we could test. Is there a way we can check the interface is complete in CI?

License: MIT
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

File: packages/client/src/lib/interface.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export interface API {
    * Stores a single file and returns a corresponding CID.
    */
   storeBlob(service: Service, content: Blob | File): Promise<CIDString>
+  /**
+   * Stores CAR file and returns a corresponding CID.
+   */
+  storeCar(service: Service, content: Blob | File): Promise<CIDString>
```
Comment (alanshaw) [inline packages/client/src/lib/interface.ts:52]: I think this takes an options object too now.

**Comment by vasco-santos** [comment]
>  Is there a way we can check the interface is complete in CI?

There is, but no clean approach. I was actually thinking about creating a module/action that could do this. Usually, we need to create a test project like https://github.com/multiformats/js-multiformats/tree/master/test/ts-use and run a CI job

File: packages/client/src/lib/interface.ts (hunk 1 above)
Comment (olizilla) [reply packages/client/src/lib/interface.ts]: like dis?

```ts
  storeCar(
    service: Service,
    content: Blob | File,
    options?: { onStoredChunk?: (size: number) => void }
  ): Promise<CIDString>
```

File: packages/client/src/lib/interface.ts [hunk 2]
```diff
@@ -46,6 +46,14 @@ export interface API {
    * Stores a single file and returns a corresponding CID.
    */
   storeBlob(service: Service, content: Blob | File): Promise<CIDString>
+  /**
+   * Stores CAR file and returns a corresponding CID.
+   */
+  storeCar(
+    service: Service,
+    content: Blob | File,
```
Comment (vasco-santos) [inline packages/client/src/lib/interface.ts:54]:
```suggestion
    content: Blob |CarReader,
```

and carReader needs to be imported

**Comment by olizilla** [comment]
Curiously, the API impls conformance to the ts definition does get tested in the build step, but it does not complain about the missing options object. I'm missing a context about what tsc is actually verifying here.

File: packages/client/src/lib/interface.ts (hunk 2 above)
Comment (olizilla) [reply packages/client/src/lib/interface.ts:54]: is?

content: Blob | File | CarReader,

in the case where you upload a CAR from a file input, you'd have a File. Though as File implements Blob, i'm not sure what benefit we get from stating that either is acceptable.

Comment (vasco-santos) [reply packages/client/src/lib/interface.ts:54]: yeah, just edited

Comment (vasco-santos) [reply packages/client/src/lib/interface.ts:54]: > in the case where you upload a CAR from a file input, you'd have a File. Though as File implements Blob, i'm not sure what benefit we get from stating that either is acceptable.

Yes, I think we can drop File

Comment (alanshaw) [reply packages/client/src/lib/interface.ts:54]: `File` helps _us_ if we want to access properties on content that only `File` implements. I think it doesn't matter for consumers.

Comment (olizilla) [reply packages/client/src/lib/interface.ts]: We don't use any File properties in the current impl, so it is removed.

**Review by alanshaw — APPROVED** [review]

**Pull request merged**